kingpin/actors/aws/ecs.py

```diff
@@ -1044,11 +1044,30 @@ def _update_service(self, service_name, existing_service, override=None):
         if override is not None:
             update_parameters.update(override)
 
+        
+        
         self.log.info('Updating service.')
-
-        yield self.api_call(
-            self.ecs_conn.update_service,
-            **update_parameters)
+        try:
+            self.log.info('Checking if existing service is active.')
+            service_is_active = (existing_service and
+                                 existing_service['status'] != 'INACTIVE')
+            if not service_is_active:
+                # We can only update an existing service.
+                self.log.error(
+                    'Could not find service with name {} to update '
+                    'in {}. Update is likely to fail.'.format(
+                        service_name, self._format_location()))
+            yield self.api_call(
+                self.ecs_conn.update_service,
+                **update_parameters)
+        except self.ecs_conn.exceptions.ServiceNotFoundException as e:
```

Comment:
My best guess at how to catch the error, from the discussion here: https://nextdoor.slack.com/archives/C03H4GRDF/p1585779522133400